src/Form/CaseFilterType.php

```diff
@@ -238,5 +244,33 @@ function (FormEvent $event) use ($formModifier) {
                 ])
             ;
         }
+
+        $builder->add('activeFilter', Filters\ChoiceFilterType::class, [
+            'choices' => [
+                $this->translator->trans('Active', [], 'case') => CaseSpecialFilterStatuses::ACTIVE,
+                $this->translator->trans('Not active', [], 'case') => CaseSpecialFilterStatuses::NOT_ACTIVE,
+            ],
+            'apply_filter' => function (QueryInterface $filterQuery, $field, $values) {
+                if (empty($values['value'])) {
+                    return null;
+                }
+
+                $filterChoice = $values['value'];
+
+                // Modify query builder according to filter choice
+                switch ($filterChoice) {
+                    case CaseSpecialFilterStatuses::ACTIVE:
+                        $filterQuery->getQueryBuilder()->andWhere($this->caseEntityRepository->getExprWithBoardFinishStatuses($filterQuery->getQueryBuilder()));
+                        break;
+                    case CaseSpecialFilterStatuses::NOT_ACTIVE:
+                        $filterQuery->getQueryBuilder()->andWhere($this->caseEntityRepository->getExprWithBoardFinishStatuses($filterQuery->getQueryBuilder(), false));
+                        break;
+                }
+
+                return $filterQuery->createCondition($filterQuery->getExpr()->andX(), []);
```

Comment:
This looks weird. Are you returning an empty condition to trick the choice filter into having an effect (cf. my comment on modifying the query builder)?

Reply:
Modified code to add conditions to `$filterQuery` instead.